dask_geopandas/morton_distance.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import pandas as pd
+from dask_geopandas.hilbert_distance import _continuous_to_discrete_coords
+
+# Based on #http://graphics.stanford.edu/~seander/bithacks.html#InterleaveBMN
+
+
+def _morton_distance(gdf, total_bounds, p):
+    """
+    Calculate distance of geoms along Morton curve
+
+    The Morton curve is also known as Z-order https://en.wikipedia.org/wiki/Z-order
+
+    Parameters
+    ----------
+    gdf : GeoDataFrame
+    total_bounds : array_like
+        array containing xmin, ymin, xmax, ymax
+    p : int
+        precision of the Morton curve
+
+    Returns
+    -------
+    type : pandas.Series
+        Series containing distances from Morton curve
+
+    """
+
+    # Calculate bounds as numpy array
+    bounds = gdf.bounds.to_numpy()
+    # Calculate discrete coords based on total bounds and bounds
+    coords = _continuous_to_discrete_coords(total_bounds, bounds, p)
+    # Calculate distance from morton curve
+    distances = _distances_from_coordinates(coords)
+
+    return pd.Series(distances, index=gdf.index, name="morton_distance")
+
+
+def _distances_from_coordinates(coords):
+    """
+    Calculate distances from geometry mid-points along Morton curve
+
+    Parameters
+    ----------
+    coords : array_like
+        x, y coordinate pairs based on mid-points of geoms
+
+    Returns
+    -------
+    type : int
+        Integer distances from Morton curve
+    """
+
+    return np.bitwise_or(
+        _part1by1(coords[:, 0]), np.left_shift(_part1by1(coords[:, 1]), 1)
+    )
+
+
+def _part1by1(n):
```

Comment:
Do you have an idea where the "part1by1" name comes from? (I mean, it comes from the active state recipe, but what's the reason for it? (I find it a bit cryptic name))

Reply:
I agree, the naming convention is a bit cryptic but seems to be widely adopted. See [`Real-Time Collision Detection book, Chapter 7`](http://www.r-5.org/files/books/computers/algo-list/realtime-3d/Christer_Ericson-Real-Time_Collision_Detection-EN.pdf
).

I think it comes from the fact the part1by1 function inserts one zero bit between each original data bit (part), the part1by2 function inserts two zero bits between each original data bit (part) and so on...